File: back-end/app/infrastructure/persistence/models/portfolio_model.py
"""
Modelo SQLAlchemy para el perfil del portfolio (fila singleton).
"""

from sqlalchemy import Column, String, DateTime, Text, CHAR
from datetime import datetime
import uuid

from ..base import Base


class PortfolioProfileModel(Base):
    """Modelo de base de datos para el perfil del portfolio."""

    __tablename__ = "portfolio_profiles"

    id = Column(CHAR(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    name = Column(String(200), nullable=False)
    title = Column(String(300), nullable=False)
    bio = Column(Text)
    avatar_url = Column(String(500))
    roles = Column(Text)  # JSON list
    stack_items = Column(Text)  # JSON list
    about_points = Column(Text)  # JSON list
    highlights = Column(Text)  # JSON list of {label,value,icon,order}
    social_links = Column(Text)  # JSON object
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, onupdate=datetime.utcnow)

    def __repr__(self) -> str:
        return f"<PortfolioProfile {self.name}>"

File: front-end/portafolio/src/app/core/models/portfolio.model.ts
export interface PortfolioSocialLinks {
  email?: string;
  github?: string;
  linkedin?: string;
  twitter?: string;
}

export interface PortfolioHighlight {
  label: string;
  value: string;
  icon?: string | null;
}

/** Contrato de GET/PUT /portfolio/profile (PortfolioProfileDTO). */
export interface PortfolioProfile {
  name: string;
  title: string;
  bio?: string | null;
  avatar_url?: string | null;
  roles: string[];
  stack_items: string[];
  about_points: string[];
  highlights: PortfolioHighlight[];
  social_links: PortfolioSocialLinks;
}

/** Contrato de POST /portfolio/avatar (201). No es PortfolioProfileDTO. */
export interface AvatarUploadResponse {
  avatar_url: string;
  id: string;
  filename: string;
  content_type: string;
  size: number;
}

/** Contrato de DELETE /portfolio/avatar. */
export interface AvatarDeleteResponse {
  avatar_url: null;
  message?: string;
}

